src/app/setup/github-oauth.component.ts

```diff
@@ -59,6 +60,16 @@ export class GitHubOAuthSetupComponent implements OnInit {
   }
 
   /**
+   * Retrieves the callback URL based on the window location HREF.
+   */
+  getCallbackUrl() {
```

Comment:
Of course, here we could do a one-liner or simple `window.location.protocol+"//"+window.location.host + "/"`, but I think keeping it human-readable and less clever is usually better. If there are any objections to that, I'm happy to change it.